src/App.tsx

```diff
@@ -1,156 +1,256 @@
-            <label className="todo__status-label">
-              <input type="checkbox" className="todo__status" />
-            </label>
-
-            <span className="todo__title">Todo is being saved now</span>
-            <button type="button" className="todo__remove">×</button>
-
-            {/* 'is-active' class puts this modal on top of the todo */}
-            <div className="modal overlay is-active">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-        </section>
-
-        {/* Hide the footer if there are no todos */}
-        <footer className="todoapp__footer">
-          <span className="todo-count">
-            3 items left
-          </span>
-
-          {/* Active filter should have a 'selected' class */}
-          <nav className="filter">
-            <a href="#/" className="filter__link selected">
-              All
-            </a>
-
-            <a href="#/active" className="filter__link">
-              Active
-            </a>
-
-            <a href="#/completed" className="filter__link">
-              Completed
-            </a>
-          </nav>
-
-          {/* don't show this button if there are no completed todos */}
-          <button type="button" className="todoapp__clear-completed">
-            Clear completed
-          </button>
-        </footer>
-      </div>
+        {todos && (
+          <TodoList
+            todos={todos}
+            updateTodo={updateTodo}
+            idTodo={idTodo}
+            deleteTodo={deleteTodo}
+          />
+        )}
+
+        {!!todosOriginal?.length && (
+          <footer className="todoapp__footer">
+            <span className="todo-count">
+              {`${activeTodo} items left`}
+            </span>
+
+            <TodoFilter
+              all={all}
+              active={active}
+              completedTodo={completedTodo}
+              statusTodosHandler={statusTodosHandler}
+              statusTodo={statusTodo}
+              setStatus={setStatus}
+            />
 
-      {/* Notification is shown in case of any error */}
-      {/* Add the 'hidden' class to hide the message smoothly */}
-      <div className="notification is-danger is-light has-text-weight-normal">
-        <button type="button" className="delete" />
+            {activeTodo !== todosOriginal.length && (
+              <button
+                type="button"
+                className="todoapp__clear-completed"
+              >
+                Clear completed
+              </button>
+            )}
+          </footer>
+        )}
+      </div>
 
-        {/* show only one message at a time */}
+      <div
+        className="notification is-danger is-light has-text-weight-normal"
+        hidden={error !== Error.add}
+      >
+        <button
+          type="button"
+          className="delete"
+          onClick={() => setError(null)}
+        />
```

Comment:
why do you have to render the same elements 3 times if you can render just different text depending on the error